fastai/callback/schedule.py

```diff
@@ -209,22 +264,36 @@ def plot_lr_find(self:Recorder, skip_end=5):
     ax.set_ylabel("Loss")
     ax.set_xlabel("Learning Rate")
     ax.set_xscale('log')
+    if suggestions:
+        colors = plt.rcParams['axes.prop_cycle'].by_key()['color'][1:]
+        for (val, idx), nm, color in zip(suggestions, nms, colors):
+            ax.plot(val, idx, 'ro', label=nm, c=color)
+        ax.legend(loc='best')
 
 # Cell
-SuggestedLRs = collections.namedtuple('SuggestedLRs', ['lr_min', 'lr_steep'])
+mk_class("SuggestionMethod", **{o.__name__.capitalize():o for o in [valley,slide,minimum,steep]},
+         doc="All possible suggestion methods as convience attributes to get tab-completion and typo-proofing")
 
 # Cell
 @patch
-def lr_find(self:Learner, start_lr=1e-7, end_lr=10, num_it=100, stop_div=True, show_plot=True, suggestions=True):
-    "Launch a mock training to find a good learning rate, return lr_min, lr_steep if `suggestions` is True"
+def lr_find(self:Learner, start_lr=1e-7, end_lr=10, num_it=100, stop_div=True, show_plot=True, suggest_funcs=(SuggestionMethod.Valley)):
```

Comment:
Would it make sense to leave the `suggestions` argument in lr_find to allow for backward compatibility?

Reply:
There's two routes we an take here:

Leave it in and have it check that both `suggest_funcs` is not `None` and that `suggestions=True`, only then having it give suggestions.

Or just raise a deprecation warning. One is more explicit than the other, and maintains the API, so I don't *think* it would be a bad thing to do

Reply:
This might be an @jph00 decision, but I feel leaving it in might make it more a complex api, as now you need to make sure that you have both suggest_funcs and set suggestions to True. So deprecating it and easing it out of the API is favored in my head

If you agree Jeremy then I can have it go and raise a depreciation warning as its set to `True`